### ISSUE TITLE
Handling of students back-tracking within a stage.

### ISSUE THREAD

**Issue by patudom** [cosmicds/educator_dashboard#52]
If this will be solved with by storing stage progress within the state, this can be closed.  

In the current dashboard that is on main as of 10/13/2023, I ran the HubbleDS as student #4226. I finished Stage 1, but tried going backwards because the notebook did the annoying thing where it cached my MC response from the previous student and I was hoping to be able to still answer it.. When I went backwards through the stage, my progress went backwards to 0.

We should be tracking the last place the student was within a stage, not where they currently are.
<img width="1049" alt="Screen Shot 2023-10-13 at 1 55 36 PM" src="https://github.com/cosmicds/educator_dashboard/assets/12750048/c26a3e55-f5ec-47a9-9b66-90caf3951eae">

**Comment by Carifio24** [comment]
Would https://github.com/cosmicds/hubbleds/pull/302 fix this?

**Comment by johnarban** [comment]
yes, https://github.com/cosmicds/hubbleds/pull/302 will fix this because it uses the max stage marker to calculate, which isn't something we previously stored in the state
